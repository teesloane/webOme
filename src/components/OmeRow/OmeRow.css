.OmeRow {
  display: flex;
  flex-direction: column;
}