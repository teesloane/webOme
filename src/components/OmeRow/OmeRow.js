import React, { PropTypes } from 'react';
import './OmeRow.css';
import OmeBtn from '../OmeBtn/OmeBtn';

const OmeRow = props => {
  // props.notes is an observable array, need to loop over it, 
  let notes = Object.keys(props.notes);
  console.log('OmeRows props are :', props)
  return (
    <main className="OmeRow">
      {notes.map(note => {
       return <OmeBtn key={note} rowId={props.rowId} noteId={note} note={props.notes[note]} />
      })}

    </main>
  );
};

OmeRow.propTypes = {
  rowId: PropTypes.string,
  notes: PropTypes.object,
};

export default OmeRow;