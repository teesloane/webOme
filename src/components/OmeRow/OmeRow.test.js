import OmeRow from './OmeRow'

describe('OmeRow Component', function () {
  beforeEach(function () {
    this.notes = {
      button_0: {
        id: 'button_0',
        midiNote: '58',
        noteOn: false,
      }
    }
  })

  it('renders', function() {
    shallow( <OmeRow notes={this.notes}/>);
  })

  it('matches snapshot', function() {
    const component = shallow( <OmeRow notes={this.notes} /> );
    expect(component).toMatchSnapshot();
  })

})
