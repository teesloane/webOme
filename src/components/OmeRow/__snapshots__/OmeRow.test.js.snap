exports[`OmeRow Component matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <main
    className="OmeRow">
    <OmeBtn
        note={
            Object {
                "id": "button_0",
                "midiNote": "58",
                "noteOn": false,
              }
        }
        noteId="button_0"
        rowId={undefined} />
</main>,
  "nodes": Array [
    <main
      className="OmeRow">
      <OmeBtn
            note={
                  Object {
                        "id": "button_0",
                        "midiNote": "58",
                        "noteOn": false,
                      }
            }
            noteId="button_0"
            rowId={undefined} />
</main>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <OmeRow
        notes={
                Object {
                        "button_0": Object {
                          "id": "button_0",
                          "midiNote": "58",
                          "noteOn": false,
                        },
                      }
        } />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "notes": Object {
            "button_0": Object {
              "id": "button_0",
              "midiNote": "58",
              "noteOn": false,
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <main
          className="OmeRow">
          <OmeBtn
                    note={
                              Object {
                                        "id": "button_0",
                                        "midiNote": "58",
                                        "noteOn": false,
                                      }
                    }
                    noteId="button_0"
                    rowId={undefined} />
</main>,
        "_debugID": 4,
        "_renderedOutput": <main
          className="OmeRow">
          <OmeBtn
                    note={
                              Object {
                                        "id": "button_0",
                                        "midiNote": "58",
                                        "noteOn": false,
                                      }
                    }
                    noteId="button_0"
                    rowId={undefined} />
</main>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <OmeRow
    notes={
        Object {
            "button_0": Object {
              "id": "button_0",
              "midiNote": "58",
              "noteOn": false,
            },
          }
    } />,
}
`;
